js/main.js

```diff
@@ -158,33 +224,42 @@ require(['gitbook', 'jQuery', 'lodash'], function (gitbook, $, _) {
     function filterThreads(section) {
         var words = section.split(' ');
 
-        return _.chain(allThreads)
-            .map(function(thread) {
-                var threadWords = (thread.context.section || '').split(' ');
+        // Compute matching for threads
+        var results = $.map(allThreads, function(thread) {
+            var threadWords = (thread.context.section || '').split(' ');
```

Comment:
Ok, forget that, I got what's going on here. We're just attaching threads to each section from which they have been created.
I stand still on the remark below about sorting though :)